### PR TITLE
Fixed links and added c++ docs per issue 4431

### LINKED ISSUE #4116: [DOC]: Fix link errors pointing to the C/C++ docs in the cuGraph site
### How would you describe the priority of this documentation request

Critical (currently preventing usage)

### Please provide a link or source to the relevant docs

https://github.com/rapidsai/cugraph/blob/branch-24.02/docs/cugraph/source/graph_support/algorithms.md

### Describe the problems in the documentation

(https://github.com/rapidsai/cugraph/blob/branch-24.02/docs/cugraph/source/graph_support/cpp/src/traversal/README.md#BFS)

This is not the correct link...all the c/C++ README links are similarly incorrect

### (Optional) Propose a correction

_No response_

### Code of Conduct

- [X] I agree to follow cuGraph's Code of Conduct
- [X] I have searched the [open documentation issues](https://github.com/rapidsai/cugraph/issues?q=is%3Aopen+is%3Aissue+label%3Adoc) and have found no duplicates for this bug report

### LINKED ISSUE #4431: [DOC]: Fix readme links on Algo support page
### How would you describe the priority of this documentation request

Critical (currently preventing usage)

### Please provide a link or source to the relevant docs

https://docs.rapids.ai/api/cugraph/nightly/graph_support/algorithms/

### Describe the problems in the documentation

Links to the C and C++ API pages are broken

### (Optional) Propose a correction

_No response_

### Code of Conduct

- [X] I agree to follow cuGraph's Code of Conduct
- [X] I have searched the [open documentation issues](https://github.com/rapidsai/cugraph/issues?q=is%3Aopen+is%3Aissue+label%3Adoc) and have found no duplicates for this bug report

### PR DESCRIPTION
**Pull request by acostadon**
added content to document c++ algorithms and fixed links that were pointing to the previously removed content
resolves #4431  
Resolves #4116

### CODE REVIEW

**Review by rlratzel — APPROVED** [review]
Looks fine structurally, but maybe @ChuckHastings should review the content for the C++ APIs.

**Review by BradReesWork — APPROVED** [review]

File: docs/cugraph/source/graph_support/algorithms/cpp_algorithms/centrality_cpp.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Centrality algorithms
+cuGraph Pagerank is implemented using our graph primitive library
+
+## Pagerank
+
+The unit test code is the best place to search for examples on calling pagerank.
+
+ * [SG Implementation](https://github.com/rapidsai/cugraph/blob/main/cpp/tests/link_analysis/pagerank_test.cpp)
+ * [MG Implementation](https://github.com/rapidsai/cugraph/blob/branch-24.06/cpp/tests/link_analysis/mg_pagerank_test.cpp)
```
Comment (ChuckHastings) [inline docs/cugraph/source/graph_support/algorithms/cpp_algorithms/centrality_cpp.md:9]: Do we want to point to a particular branch?  That seems like we would need to update every release.

File: docs/cugraph/source/graph_support/algorithms/cpp_algorithms/traversal_cpp.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# Traversal
+cuGraph traversal algorithms are contained in this directory
+
+## SSSP
+
+The unit test code is the best place to search for examples on calling SSSP.
+
+ * [SG Implementation](https://github.com/rapidsai/cugraph/blob/branch-24.06/cpp/tests/traversal/sssp_test.cpp)
```
Comment (ChuckHastings) [inline docs/cugraph/source/graph_support/algorithms/cpp_algorithms/traversal_cpp.md:8]: Same question about the specific version/branch

**Review by ChuckHastings — APPROVED** [review]

**Comment by acostadon** [comment]
rerun tests

**Comment by acostadon** [comment]
rerun tests

**Pull request merged**